Get tailscale tokens via vault instead

lheckemann opened the pull request.

lheckemann committed:
Get tailscale token from Vault instead of Mosyle

lheckemann committed:
note for foundation macs so we don't forget

lheckemann committed:
Use "main" token, not SSH host key signer token, to acquire tailscale key

lheckemann committed:
vault: apply tag when setting up tailscale, and only set it up on detsys macs

lheckemann committed:
They don't

cole-h commented on setup-vault.sh:
```diff
@@ -86,6 +86,10 @@ set -o pipefail
     unset SECRET_ID_FILE
     unset ROLE_ID
+    if ! grep -q foundation "$ROLE_ID_FILE" ; then
+      $VAULT read -field=key internalservices/macos/tailscale/key tags=tag:ephemeral-mac-ci > /nix/home/tailscale.token
```
```suggestion
      $VAULT read -field=key internalservices/macos/tailscale/key ephemeral=true tags=tag:ephemeral-mac-ci > /nix/home/tailscale.token
```

cole-h reviewed (commented):
LGTM.

However, the vault plugin doesn't (yet) support ephemeral tokens (see https://github.com/davidsbond/vault-plugin-tailscale/pull/51), so we'll want to prod upstream for a new release with that merged before we can actually use this, I think.

We might also need to request the "reusable" setting, unless that was just to work around the lack of ephemerality in the token(s) themselves (which we now have as a result of the vault backend).

lheckemann committed:
debugging

lheckemann committed:
add tailscale to systemPackages

This resolves an inconsistency between non-nix and nix macs.

lheckemann committed:
use ephemeral

lheckemann commented:
>we'll want to prod upstream for a new release with that merged before we can actually use this, I think.

With the power of Nix, we can just pull in the commit implementing it ;)

>We might also need to request the "reusable" setting

This shouldn't be necessary:
>[Ephemeral devices are auto-removed anywhere from 30 minutes to 48 hours after the last activity](https://tailscale.com/kb/1111/ephemeral-nodes/#how-long-before-ephemeral-devices-are-auto-removed)

That's easily enough for e.g. rebooting a node or the router, and if it does come to it and one of the devices loses its access to the net we can just bonk it again.

lheckemann committed:
Fix timeout on waiting for SSH host key

lheckemann committed:
Use /var/root/tailscale.token consistently for the tailscale token

lheckemann committed:
Ensure tailscale token isn't world-readable

lheckemann committed:
Improve tailscale-auth daemon on non-nix machines

lheckemann committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'darwin':
    'github:LnL7/nix-darwin/ef56fd8979b5f4e800c4716f62076e00600b1172' (2022-10-10)
  → 'github:LnL7/nix-darwin/87b9d090ad39b25b2400029c64825fc2a8868943' (2023-01-09)
• Updated input 'nixpkgs':
    'github:nixos/nixpkgs/93e0ac196106dce51878469c9a763c6233af5c57' (2022-10-21)
  → 'github:nixos/nixpkgs/49596eb4e50b18337aad3b73317371bcd4e3b047' (2023-03-06)

cole-h commented on README.md:
````diff
@@ -316,26 +316,6 @@ Note that right now this code assumes you're installing everything for DetSys pu
 It is an explicit goal for this repository to support configuring things for *your* purposes without necessarily having to fork the repo.
 Please open issues discussing or send PRs improving this.
 
-#### Tailscale Token
-
-The tailscale token should be configured as follows.
-
-First configure a tag to assign to ephemeral macs, by adding this to your ACL:
-
-```json
-	"tagOwners": {
-		"tag:ephemeral-mac-ci": ["you@example.com"],
-	}
-```
````
We probably want to keep this as documentation.

lheckemann replied:
This token now comes from Vault, so there's nothing left to _actually do_ for the setup here -- and this was quite clearly "task" documentation and not "how it works" documentation.

cole-h replied:
I think that's not exactly true, and we do still want this: https://tailscale.com/kb/1068/acl-tags/#defining-a-tag says that you need to define the tag before you can use it (and somewhat relatedly, tagged machines don't expire since March 2022: https://tailscale.com/blog/tagged-key-expiry/).

lheckemann committed:
Restore/update tailscale config readme section
